docs/sql_permissions.md

```diff
@@ -0,0 +1,89 @@
+# SQL Permissions
+
+Promscale adopts a Role Based Access Control (RBAC) philosophy to managing permissions
+at the SQL layer. Many users will choose to simply use a single PostgreSQL user for all
+Promscale operations (using a single owner user as defined belows).
+But we also allow users more granular control over permissions management.
+
+Promscale defines several roles:
+- **Owner** this is the owner of the Promscale schema and objects. It is defined
+  as the user that originally installed/migrated Promscale. This user needs to be
+  able to install several PostgreSQL extensions and thus either needs to be a
+  PostgreSQL superuser or use [pgextwlist](https://github.com/dimitri/pgextwlist) to authorize
+  a non-superuser to install the extensions(see below). The owner should be consistent
+  every time you migrate the Promscale schema.
+- **prom_reader** this role is allowed to read but not modify the data stored by Promscale.
+- **prom_writer** is allowed to read Promscale data as well as write (insert) new data. This
+  role is not allowed to modify or delete data. Includes all permissions of prom_reader.
+- **prom_modifier** is able to read, write, and modify all Promscale data. Includes all permissions of prom_writer.
+- **prom_admin** is allowed to change the configuration options associated with Promscale. This
+  includes data retention policies, chunk intervals, etc. Includes all permissions of prom_modifier.
+- **prom_maintenance** is allowed to execute maintenance tasks such as compression and data retention jobs.
+  mostly only used externally unless using a CRON job to execute_maintenance() instead of the jobs framework.
+  Includes all permissions of prom_reader.
+
+You can assign roles to users with the following sql command:
+`GRANT <role> to <user>`. A user can be assigned multiple roles. More information
+can be found in the [PostgreSQL docs about permissions](https://www.postgresql.org/docs/current/user-manag.html).
+
+
+Promscale can be run by the owner or a user with the `prom_modifier`, `prom_writer` or `prom_reader` role.
+The following table describes what features of promscale are available with each role:
+
+|user / role| schema migration | delete series api | write api | read api |
+| --- | --- | --- | --- | --- |
+| owner | ✅ | ✅ | ✅ | ✅|
```

Comment:
should prom_admin be in this table? Is it any different than modifier? Or is prom_admin the owner user? If so, why the two names? Probably want to add a note about what the admin user is for if it's not in this hierarchy.

Reply:
Found the note above, sorry. Missed that, that definitely covers that bit. I do think if you shouldn't run it as admin for some reason you should put a note here, but it looks like it would work...